test/benchmarks/polynomial_dfs_benchmark.cpp

```diff
@@ -0,0 +1,215 @@
+#include <boost/algorithm/string.hpp>
+#include <boost/test/unit_test.hpp>
+#include <boost/timer/progress_display.hpp>
+#include <boost/timer/timer.hpp>
+
+#include <nil/crypto3/algebra/fields/arithmetic_params/bls12.hpp>
+#include <nil/crypto3/math/polynomial/polynomial_dfs.hpp>
+#include <nil/crypto3/random/algebraic_engine.hpp>
+
+
+// Benchmark test cases integrated to Boost.Test framework, check the examples below
+struct test_case_base {
+    using MeanQuantileAccumulatorSet = boost::accumulators::accumulator_set<
+        double,
+        boost::accumulators::features<
+            boost::accumulators::tag::mean,
+            boost::accumulators::tag::extended_p_square_quantile
+        >
+    >;
+
+    std::map<std::string, boost::timer::cpu_timer> timers;
+    std::map<std::string, MeanQuantileAccumulatorSet> accumulators;
+    std::vector<double> probs = {0.5, 0.9, 0.95, 0.99};
+
+    void run_benchmark_iterations(
+        int num_iterations,
+        std::function<void()> benchmark_impl
+    ) {
+        boost::timer::progress_display progress_bar(num_iterations);
+        for (int i = 0; i < num_iterations; ++i) {
+            benchmark_impl();
+            for (const auto& [flag, timer] : timers) {
+                auto acc = accumulators.emplace(
+                    std::piecewise_construct,
+                    std::forward_as_tuple(flag),
+                    std::forward_as_tuple(boost::accumulators::extended_p_square_probabilities = probs)
+                );
+                acc.first->second(timer.elapsed().wall * 1.0e-9);
+            }
+            timers.clear();
+            ++progress_bar;
+        }
+    }
+
+    void report_results() {
+        using namespace boost::accumulators;
+        for (const auto& acc : accumulators) {
+            std::cout << "Results for " << acc.first << ":\n"
+                << " Mean time: " << std::fixed << std::setprecision(3) << mean(acc.second) << " seconds\n"
+                << " Percentiles:\n" << std::fixed;
+            for (auto prob : probs) {
+                std::cout << "  " << std::setprecision(0) << prob * 100 << "th: "
+                    << std::setprecision(3) << quantile(acc.second, quantile_probability = prob) << " seconds\n";
+            }
+            std::cout << "\n";
+        }
+    }
+};
+
+#define BENCHMARK_FIXTURE_TEST_CASE(test_case_name, num_iterations, fixture) \
+    struct test_case_name : public fixture, test_case_base {                 \
+        void test_method();                                                  \
+    };                                                                       \
+    static void BOOST_AUTO_TC_INVOKER( test_case_name )()                    \
+    {                                                                        \
+        test_case_name t;                                                    \
+        t.run_benchmark_iterations(                                          \
```

Comment:
Let's add a warmup stage here, like call the same function num_iterations / 10 times, before turning on the timer. I'm not sure why, but looks like in these cases we get more stable results, standard deviation of results becomes 0 that way.